crates/circuit/src/dag_circuit.rs

```diff
@@ -5074,7 +5074,7 @@ impl DAGCircuit {
     /// This is mostly used to apply operations from one DAG to
     /// another that was created from the first via
     /// [DAGCircuit::copy_empty_like].
-    fn push_back(&mut self, py: Python, instr: PackedInstruction) -> PyResult<NodeIndex> {
+    pub fn push_back(&mut self, py: Python, instr: PackedInstruction) -> PyResult<NodeIndex> {
```

Comment:
This should go back to being private, use `DAGCircuit::apply_operation_back` for future reference.

```suggestion
    fn push_back(&mut self, py: Python, instr: PackedInstruction) -> PyResult<NodeIndex> {
```

Reply:
I know you are better up to date with this topic, so I wonder what's your view on https://github.com/Qiskit/qiskit/pull/13141#discussion_r1782441466. It's the reason why I didn't revert this change, together with a couple of other examples scattered around the code, and I wonder if it might be worth exposing this method?